Makefile

```diff
@@ -1,8 +1,3 @@
-# Image URL to use all building/pushing image targets
```

Comment:
I refactored Makefile to keep only instructions that we use today.
We can add more of them, if we require them in the future.

Also, I install envtest-setup and controller-gen to the local bin, similar like in [other k8s projects](https://github.com/kubernetes-sigs/jobset/blob/main/Makefile#L250).

@kubeflow/wg-training-leads Please let me know if that looks good.